Windows

RomanK17 opened the pull request.

RomanK17 committed:
add modals

RomanK17 committed:
add

RomanK17 committed:
dlt index.html from PR

RomanK17 committed:
add index.html

andrey-kudinov commented on src/js/modules/modals.js:
```diff
@@ -1 +1,67 @@
+const modals = () => {
+    function bindModal(triggerSelector, modalSelector, closeSelector) {
+
+        const trigger = document.querySelectorAll(triggerSelector),
+            modal = document.querySelector(modalSelector),
+            close = document.querySelector(closeSelector);
```
Старайся пж не использовать синтаксис чз запятую
Считается, что это сложнее для чтения кода

andrey-kudinov commented on src/js/modules/modals.js:
```diff
@@ -1 +1,67 @@
+
+const modals = () => {
+    function bindModal(triggerSelector, modalSelector, closeSelector) {
+
```
Удали пж лишний вертикальный отступ

andrey-kudinov commented on src/js/modules/modals.js:
```diff
@@ -1 +1,67 @@
 'use strict';
```
По дефолту в ESM 'use strict' поэтому эту строчку можно удалить

andrey-kudinov commented on src/js/modules/modals.js:
```diff
@@ -1 +1,67 @@
+const modals = () => {
+    function bindModal(triggerSelector, modalSelector, closeSelector) {
+
+        const trigger = document.querySelectorAll(triggerSelector),
```
Хочу обратить твоё внимание на нейминг
В твоём случае здесь массив элементов, поэтому trigger**s**

andrey-kudinov commented on src/js/modules/modals.js:
```diff
@@ -1 +1,67 @@
+            document.body.style.overflow = 'hidden';
+        }
+
+        trigger.forEach((btn) => {
```
Здесь так же хотел бы обратить внимание на нейминг
Старайся пж придерживаться более говорящего неймига
Тогда код будет более читабельный
На сложный проектах это играет большую роль
```
triggers.forEach(trigger => ...
cards.forEach(card => ...
books.forEach(book => ...
```

andrey-kudinov commented on src/js/modules/modals.js:
```diff
@@ -1 +1,67 @@
+
+    }
+
+    function showModalByTime(selector, time) {
```
Давай попробуем в этом проекте писать исключительно стрелочные функции
Во-первых, практика рефакторинга (довольно простого)
Во-вторых, на мой взгляд, это позволит лучше читать код, так как он будет последовательный
Всегда понятно что за функция будет вызвана, так как она до этого объявлена
Так же на мой взгляд это современнее

andrey-kudinov commented on src/js/modules/modals.js:
```diff
@@ -1 +1,67 @@
+    }
+
+    function showModalByTime(selector, time) {
+        setTimeout(() => {
```
`modalTimer = setTimeout(() => {`

andrey-kudinov replied:
а до этого на верхнем уровне создай `let modalTimer`

andrey-kudinov replied:
Тогда ты можешь в нужный момент использовать `clearTimeout(modalTimer);`

andrey-kudinov reviewed (changes requested):
Хорошая работа 👍
Есть несколько моментов, которые можно улучшить
Посмотри пж в комментариях

RomanK17 committed:
some minor changes according to the comments

Add:
changed some syntax, names of variables;
replaced all common functions with arrow functions;
fixed problem with timer (It appeared even if the modal window was already called).
